index.js

```diff
@@ -0,0 +1,267 @@
+        };
+    } else
+        return null;
+}
+
+const getTextLength = function (parent, node, offset) {
+    var textLength = 0;
+
+    if (node.nodeName == '#text')
+        textLength += offset;
+    else for (var i = 0; i < offset; i++)
+        textLength += getNodeTextLength(node.childNodes[i]);
+
+    if (node != parent)
+        textLength += getTextLength(parent, node.parentNode, getNodeOffset(node));
+
+    return textLength;
+}
+
+const getNodeTextLength = function (node) {
+    var textLength = 0;
+
+    if (node.nodeName == 'BR')
+        textLength = 1;
+    else if (node.nodeName == '#text')
+        textLength = node.nodeValue.length;
+    else if (node.childNodes != null)
+        for (var i = 0; i < node.childNodes.length; i++)
+            textLength += getNodeTextLength(node.childNodes[i]);
+            
+    return textLength;
+}
+
+const getNodeOffset = function (node) {
+    return node == null ? -1 : 1 + getNodeOffset(node.previousSibling);
+}
+
+window.onload = function () {
+    hamburger.addEventListener('click',toggleList);
+    textArea.addEventListener('keydown',updateTextDown)
+    textArea.addEventListener('keyup',updateTextUp)
+    editor.addEventListener('keydown',updateEditorDown)
+    editor.addEventListener('keyup',updateEditorUp)
+    save.addEventListener('click',saveNote)
+    del.addEventListener('click',delNote)
+    addNote.addEventListener('click',addNewNote)
+    listWrapper.addEventListener('click',active)
+}
+
+setNotes()
+renderNotes()
+setQuerySelectors()
+
+function toggleList(){
+    toggleCount++;
+    leftSideBar.setAttribute("id", toggleCount % 2 === 0 ? "toggle-none-sm" : " ");
+    textareaEditorWrapper.setAttribute("id", toggleCount % 2 === 0 ? " ":"toggle-none-sm" );
+}
+
+function updateTextDown(e){
+    textAreaContent.innerText =  (e.target.value !== '' ) ? 
+        e.target.value : '.'
+    if(e.key === 'Enter'){
+        e.preventDefault()
+        editor.focus()
+    }
+    if(e.key === 'ArrowDown' || e.key === 'ArrowRight'){
+        (e.target.selectionEnd === e.target.value.length) && editor.focus() 
+    }
+}
+
+function updateTextUp(e){
+    heading.innerText =     e.target.value.length === 0 ?  'Untitled': e.target.value
+    notes[curNode.dataset.num].title = e.target.value
+}
+
+function updateEditorDown(e){
+    const textSelection = getTextSelection(document.activeElement);
+    if(textSelection.start === 0){
+        if(e.key === 'ArrowUp' || e.key === 'ArrowLeft') {
+            e.preventDefault()   
+            textArea.focus()
+        }   
+    }
+}
+
+function updateEditorUp(e){
+    span.textContent =  e.target.innerText
+}
+
+function saveNote(){
+    const curTime = new Date().toLocaleTimeString()  
+    date.innerText = `last edited ${curTime}`
+    const getCurNodeDataset = curNode.dataset.num
+    const isIndex = num => num.index == getCurNodeDataset 
+    let indexOfCurNode = notes.findIndex(isIndex) 
```

Comment:
avoid using non-recognizable short forms. It makes your code more readable by others and also you will understand better 6 months down the line :)
`cur` isn't a common shortform of `current`. So write in full `indexOfCurrentNode`